charts/vigil/templates/deployment.yaml

```
@@ -0,0 +1,71 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "vigil.fullname" . }}
+  labels:
+    {{- include "vigil.labels" . | nindent 4 }}
+spec:
+{{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "vigil.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/secrets.yaml") . | sha256sum }}
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "vigil.selectorLabels" . | nindent 8 }}
+    spec:
+      volumes:
+        - name: vigil
+          configMap:
+            name: {{ include "vigil.fullname" . }}
+            defaultMode: 0777
```

```suggestion
            readOnly: true
```

the app doesn't need to write anything there, it's just pointing to the configuration file